src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java

```diff
@@ -44,66 +49,237 @@ public void setUp() {
+        // Alice is associated with Teacher and is booked for this event => Alice should be present
+        assertTrue(teacherAlice.isPresent());
+        Optional<UserSummaryDTO> teacherCharlie = (Optional<UserSummaryDTO>) teacherEntity.stream().filter(e -> ((EventBookingDTO) e).getUserBooked().getId() == 9).findFirst();
+        // Charlie is not associated with Teacher and is not booked for this event => Charlie should not be present
+        assertFalse(teacherCharlie.isPresent());
+
+        // Try logging in with another teacher account and see if we are sending the wrong information to the wrong teachers
+        LoginResult daveLogin = loginAs(httpSession, "dave-teacher@test.com", Constants.TEST_TEACHER_PASSWORD);
+        HttpServletRequest dave_Request = createRequestWithCookies(new Cookie[] { daveLogin.cookie });
+        replay(dave_Request);
+        Response dave_Response = eventsFacade.getEventBookingForAllGroups(dave_Request, "_regular_test_event");
+        assertEquals(Response.Status.OK.getStatusCode(), dave_Response.getStatus());
+        assertNotNull(dave_Response.getEntity());
+        // Make sure the EventBookingDTOs contain UserSummaryDTOs, thus not leaking information
+        assertTrue(dave_Response.getEntity() instanceof List); // instanceof is OK here because we just need to know this is a subclass of a List
+        List<?> daveEntity = (List<?>) dave_Response.getEntity();
+        for (Object o : daveEntity) {
+            assertEquals(EventBookingDTO.class.getCanonicalName(), o.getClass().getCanonicalName());
+            assertEquals(UserSummaryDTO.class.getCanonicalName(), ((EventBookingDTO) o).getUserBooked().getClass().getCanonicalName());
+        }
+        Optional<UserSummaryDTO> daveAlice = (Optional<UserSummaryDTO>) daveEntity.stream().filter(e -> ((EventBookingDTO) e).getUserBooked().getId() == 7).findFirst();
+        // Alice is not associated with Dave but is booked for this event => Alice should not be present
+        assertFalse(daveAlice.isPresent());
+        Optional<UserSummaryDTO> daveCharlie = (Optional<UserSummaryDTO>) daveEntity.stream().filter(e -> ((EventBookingDTO) e).getUserBooked().getId() == 9).findFirst();
+        // Charlie is associated with Dave and is not booked for this event => Charlie should not be present
+        assertFalse(daveCharlie.isPresent());
+    }
+
+    @Test
+    // GET /events/{event_id}/bookings/for_group/{group_id} -> EventFacade::getEventBookingForGivenGroup(request, eventId, groupId)
+    public void getEventBookingsByEventIdForGroup() throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException, AuthenticationProviderMappingException, IncorrectCredentialsProvidedException, AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException, MFARequiredButNotConfiguredException {
```

Comment:
[In the last PR](https://github.com/isaacphysics/isaac-api/pull/452#discussion_r930003423) you mentioned you tried a few ways of structuring the tests and settled on this. My preference would be more similar to how our unit tests are (should be?) structured - that is, one test method per behaviour in a particular context (i.e. using endpoint A user B can do C when D) with clear arrange/act/assert stages.

That might result in some duplication and may be inconvenient for tests which build up cumulative state, but I think it would be more obvious what's happening.

Reply:
Yes, I've been wrestling with my decision ever since and I haven't settled fully on this style. I'm tempted to leave these tests as they are for the moment and we can try the other approach going forward, see how it feels. The risk is that we mix styles but I'll try to keep on top of this and monitor the situation.